### PR TITLE
Update 0.4.0 with contiguous split workaround

### PR DESCRIPTION
**Pull request by jlowe**
Merging in the contiguous split workaround for the 0.4.0 release.

**NOTE: This needs to be merged with a merge commit, not squashed!**

### CODE REVIEW

**Comment by jlowe** [comment]
build

**Comment by jlowe** [comment]
Fixing this in #1875

**Review by abellina — APPROVED** [review]

File: sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuCoalesceBatches.scala [hunk 1]
```diff
@@ -608,13 +608,15 @@ case class GpuCoalesceBatches(child: SparkPlan, goal: CoalesceGoal)
...
-    val cannotSpill = child.schema.fields.exists { f =>
-      f.dataType match {
-        case MapType(_, _, _) | ArrayType(_, _) | StructType(_) => true
-        case _ => false
-      }
-    }
-
+    // disabling spillable batches due to
+    // https://github.com/rapidsai/cudf/issues/7514
+    val cannotSpill = true
```
Comment (gerashegalov) [inline sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuCoalesceBatches.scala:613]: we can also consider leaving the code in but adding a new conf like "disable.spillable.batches" . We would flip the default to false once the bug is fixed.

Comment (jlowe) [reply sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuCoalesceBatches.scala:613]: IMO configs are overkill for this.  Bugfix is already posted in cudf-0.19, will definitely be fixed in 0.5.

**Review by gerashegalov — APPROVED** [review]

**Pull request merged**